test/e2e/scale.go

```diff
@@ -495,3 +579,45 @@ type workResult struct {
 	clusterName string
 	err         any
 }
+
+func addMachineDeployments(baseClusterTemplateYAML []byte, count int) []byte {
```

Comment:
As suggested above let's rewrite this a bit.

* Let's enforce that the cluster-template always has exactly 1 MD topology
* Let's ensure this func modifies the Cluster.spec.topology.workers so we end up with exactly the count we pass in 
  * This would also include dropping the existing MD topology when the count is 0